### PR TITLE
feat: add CRUD, like/dislike & comments for posts

### PR DESCRIPTION
**Pull request by anik31**
## Added - 
- CRUD for posts
- like/dislike
- comments

### Use this link - [Link for files](https://github.com/anik31/vacay-book/pull/6/files) (Doesn't contain deleted files)

### CODE REVIEW

**Review by BalajiHarinath — APPROVED** [review]
Everything looks good. Could not find any suggestions. Good that you have used Chakra UI and redux. UI is great as well.

File: src/features/post/components/CreatePost.jsx [hunk 1]
```diff
@@ -9,13 +9,36 @@ import {
...
+            dispatch(createPost({content: postInput}));
+            onClose();
+        }else{
+            toast({
+                title: "Post text is empty",
+                status: "warning",
+                position: "top-right",
+                isClosable: true,
+                duration: 3000
+            })
```
Comment (Maianki) [inline src/features/post/components/CreatePost.jsx:39]: Suggestion: you can set common props for toast in toast container itself so you do not have to pass it everytime you are using toast.

File: src/features/post/components/DisplayPost.jsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import { useState } from 'react';
+import { HamburgerIcon } from '@chakra-ui/icons';
+import { CreatePost } from './CreatePost';
+
+export function DisplayPost({post}){
+  const {_id, comments, content, createdAt, likes: {likeCount, likedBy}, username} = post;
+    const {allUsers} = useSelector(store=>store.user);
+    const {user} = useSelector(store=>store.auth);
+    const {firstName, lastName, profilePic} = allUsers.find(user=>user.username===username);
+    const fullName = firstName + " " + lastName;
```
Comment (Maianki) [inline src/features/post/components/DisplayPost.jsx]: Nit pick: Following ES6 you can use template literal here

**Review by Maianki — APPROVED** [review]
Great job. Everything looks good just added a few suggestions.

File: src/features/post/components/CreatePost.jsx (hunk 1 above)
Comment (anik31) [reply src/features/post/components/CreatePost.jsx:39]: Nice suggestion, I will look into it after completing basic features

File: src/features/post/components/DisplayPost.jsx (hunk 2 above)
Comment (anik31) [reply src/features/post/components/DisplayPost.jsx]: done

**Pull request merged**